### PR TITLE
ENH: Enable more Arrow CSV tests/features

### PR DESCRIPTION
**Pull request by lithomas1**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/whatsnew/v2.0.0.rst [hunk 1]
```diff
@@ -1283,6 +1283,7 @@ I/O
 - Bug in :meth:`DataFrame.to_dict` not converting ``NA`` to ``None`` (:issue:`50795`)
 - Bug in :meth:`DataFrame.to_json` where it would segfault when failing to encode a string (:issue:`50307`)
 - Bug in :func:`read_xml` where file-like objects failed when iterparse is used (:issue:`50641`)
+- Fixed handling of ``encoding`` and ``decimal`` parameters when ``engine="pyarrow"`` in :func:`read_csv` (:issue:`51302`)
```
Comment (phofl) [inline doc/source/whatsnew/v2.0.0.rst:1286]: Can we put decimal under enhancements?

**Comment by phofl** [comment]
I'll take a look tomorrow

**Comment by lithomas1** [comment]
gentle ping @phofl.

File: pandas/tests/io/parser/test_encoding.py [hunk 2]
```diff
@@ -20,9 +20,9 @@
 import pandas._testing as tm
 
 skip_pyarrow = pytest.mark.usefixtures("pyarrow_skip")
+xfail_pyarrow = pytest.mark.usefixtures("pyarrow_xfail")
```
Comment (phofl) [inline pandas/tests/io/parser/test_encoding.py:23]: How many tests are this? xfail kinda kills test runtime compared to skip

File: doc/source/whatsnew/v2.0.0.rst [hunk 3]
```diff
@@ -1293,6 +1294,7 @@ I/O
 - Bug in :meth:`DataFrame.to_json` where it would segfault when failing to encode a string (:issue:`50307`)
 - Bug in :meth:`DataFrame.to_html` with ``na_rep`` set when the :class:`DataFrame` contains non-scalar data (:issue:`47103`)
 - Bug in :func:`read_xml` where file-like objects failed when iterparse is used (:issue:`50641`)
+- Fixed handling of ``encoding`` parameter when ``engine="pyarrow"`` in :func:`read_csv` (:issue:`51302`)
```
Comment (phofl) [inline doc/source/whatsnew/v2.0.0.rst:1297]: Can you rephrase like the others?

File: doc/source/whatsnew/v2.0.0.rst [hunk 4]
```diff
@@ -280,6 +280,7 @@ Other enhancements
 - :func:`to_datetime` now accepts ``"mixed"`` as an argument to ``format``, which will infer the format for each element individually (:issue:`50972`)
 - Added new argument ``engine`` to :func:`read_json` to support parsing JSON with pyarrow by specifying ``engine="pyarrow"`` (:issue:`48893`)
 - Added support for SQLAlchemy 2.0 (:issue:`40686`)
+- Support ``decimal`` parameter when ``engine="pyarrow"`` in :func:`read_csv` (:issue:`51302`)
```
Comment (phofl) [inline doc/source/whatsnew/v2.0.0.rst:283]: Can you also rephrase like the others?

File: pandas/tests/io/parser/test_encoding.py (hunk 2 above)
Comment (lithomas1) [reply pandas/tests/io/parser/test_encoding.py:23]: It's like 10. The input data is tiny so it shouldn't matter. There is still a hanging test, so we still have ``skip_pyarrow`` in there.

Comment (phofl) [reply pandas/tests/io/parser/test_encoding.py:23]: ok should be fine then

**Comment by phofl** [comment]
thx @lithomas1
